CommonTasks/DscResources/SqlServer/SqlServer.schema.psm1

```diff
@@ -10,7 +10,59 @@
     Import-DscResource -ModuleName PSDesiredStateConfiguration
     Import-DscResource -ModuleName SqlServerDsc
 
-    if( $Setup -ne $null ) {
+    if( $Setup -ne $null )
+    {
+        # Remove Case Sensitivity of ordered Dictionary or Hashtables
+        $Setup = @{}+$Setup
+
+        # FileStreamAccessLevel is not supported by SqlSetup and must be enabled later
+        $fileStreamAccessLevel = $Setup.FileStreamAccessLevel
+        $Setup.Remove( 'FileStreamAccessLevel' )
+
         (Get-DscSplattedResource -ResourceName SqlSetup -ExecutionName "sqlSetup" -Properties $Setup -NoInvoke).Invoke($Setup)
+
+        # enable/disable FileStream
+        if( $null -ne $fileStreamAccessLevel )
+        {        
+            [string]$instanceName = $Setup.InstanceName
+
+            Script sqlFileStreamAccess
+            {
+                TestScript = 
+                {
+                    # get installed SQL Server version
+                    [string]$cmgmt = (Get-WmiObject -NameSpace 'ROOT\Microsoft\SQLServer' -Class “__NAMESPACE” | Where-Object { $_.Name.StartsWith( 'ComputerManagement' ) }).Name
+
+                    $wmi = Get-WmiObject -Namespace "ROOT\Microsoft\SqlServer\$cmgmt" -Class FilestreamSettings | Where-Object {$_.InstanceName -eq $using:instanceName}
+                    
+                    Write-Verbose "The current access level of FILESTREAM is set to $($wmi.AccessLevel) and the file share name is '$($wmi.ShareName)'."
+                    Write-Verbose "Expected access level of FILESTREAM is $using:fileStreamAccessLevel."
+                    
+                    if( $wmi.AccessLevel -eq $using:fileStreamAccessLevel )
+                    {
+                        return $true  
+                    }
+
+                    return $false
+                }
+                SetScript = 
+                {
+                    # get installed SQL Server version
+                    [string]$cmgmt = (Get-WmiObject -NameSpace 'ROOT\Microsoft\SQLServer' -Class “__NAMESPACE” | Where-Object { $_.Name.StartsWith( 'ComputerManagement' ) }).Name
+
+                    $wmi = Get-WmiObject -Namespace "ROOT\Microsoft\SqlServer\$cmgmt" -Class FilestreamSettings | Where-Object {$_.InstanceName -eq $using:instanceName}
```

Comment:
Can we change that from 'Get-WmiObject' to 'Get-CimInstance'? 'Get-WmiObject' uses WMI1 which was replaced in PowerShell 3 or 4.